plugins/inputs/hystrix_stream/hystrix_stream.go

```diff
@@ -0,0 +1,127 @@
+package hystrix_stream
+
+import (
+	"bufio"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  ## Hystrix stream servlet to connect to (with port and full path)
+  url = "http://localhost:8090/hystrix"
+ `
+
+type HystrixData struct {
+	Url           string
+	healthy       bool
+	scanner       *bufio.Scanner
+	cachedEntries []HystrixStreamEntry
+	reader        io.ReadCloser
+	cacheLock     sync.Mutex
+	client        *http.Client
+	Timeout       internal.Duration
+	tls.ClientConfig
+}
+
+func (s *HystrixData) Description() string {
+	return "Plugin to parse a hystrix-stream-servlet."
+}
+
+func (s *HystrixData) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *HystrixData) Gather(acc telegraf.Accumulator) error {
+
+	if s.client == nil {
+		tlsCfg, err := s.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+		s.client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: tlsCfg,
+				Proxy:           http.ProxyFromEnvironment,
+			},
+			Timeout: s.Timeout.Duration,
+		}
+	}
+	entries, errors := s.latestEntries()
+	if errors != nil {
+		return errors
+	}
+	for _, entry := range entries {
+		acc.AddFields(fieldsFromEntry(entry))
+	}
+	return nil
+}
+
+func fieldsFromEntry(entry HystrixStreamEntry) (string, map[string]interface{}, map[string]string, time.Time) {
+	counterName := entry.Group + entry.Name
+	fields := getCounterFields(entry)
+	tags := getTags(entry)
+	entryTime := time.Unix(0, entry.CurrentTime*int64(time.Millisecond))
+	return counterName, fields, tags, entryTime
+}
+
+func getTags(entry HystrixStreamEntry) map[string]string {
+	tags := make(map[string]string)
```

Comment:
Consider simply returning an anonymous map for `getTags` and `getCounterFields`:
```go
	return map[string]string{
		"name": entry.Name,
		"type": entry.Type,
		"group": entry.Group,
		"threadpool": entry.ThreadPool,
	}
```